src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -377,7 +370,9 @@ private boolean doSaveImpl(final String path,
       // Don't auto-save when there are no changes. In addition to being
       // wasteful, it causes the server to think the document is dirty.
       if (path == null && fileType == null && diff.isEmpty()
-          && foldSpec.equals(oldFoldSpec) 
+          && newFoldSpec.equals(oldFoldSpec)
+          && newSelectionSpec.equals(oldSelectionSpec)
```

Comment:
It's not the overhead of the RPCs I'm worried about (they are indeed differential) but the disk I/O on the server. An autosave causes the entire document contents to be written to disk (even though only the diff travels over the wire). This change adds a lot of disk activity which is triggered by nothing other than the user moving their cursor around. It might not be a big deal (we already tolerate this while typing) but worth keeping an eye on.